containers/unit_tests/TestDynRankView_TeamScratch.hpp

```diff
@@ -0,0 +1,70 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_DynRankView.hpp>
+
+namespace Test {
+
+void test_dyn_rank_view_team_scratch() {
+  using execution_space = TEST_EXECSPACE;
+  using memspace_type   = execution_space::scratch_memory_space;
+  using drv_type        = Kokkos::DynRankView<int, memspace_type>;
+  using policy_type     = Kokkos::TeamPolicy<execution_space>;
+  using team_type       = policy_type::member_type;
+
+  int N0 = 10, N1 = 4, N2 = 3;
+  size_t shmem_size = drv_type::shmem_size(N0, N1, N2);
+  ASSERT_GE(shmem_size, N0 * N1 * N2 * sizeof(int));
+
+  Kokkos::View<unsigned, execution_space, Kokkos::MemoryTraits<Kokkos::Atomic>>
+      errors("errors");
+  auto policy = policy_type(1, Kokkos::AUTO)
+                    .set_scratch_size(0, Kokkos::PerTeam(shmem_size));
+  Kokkos::parallel_for(
+      policy, KOKKOS_LAMBDA(const team_type& team) {
+        drv_type scr(team.team_scratch(0), N0, N1, N2);
+        // Control that the code ran at all
+        if (scr.rank() != 3) errors() |= 1u;
+        if (scr.extent(0) != N0) errors() |= 2u;
```

Comment:
Fine by me but I expected the compiler would complain about comparing signed and unsigned integers here